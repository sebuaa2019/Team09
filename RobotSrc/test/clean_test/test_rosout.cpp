#include <ros/ros.h>
#include <move_base_msgs/MoveBaseAction.h>
#include <actionlib/client/simple_action_client.h>
#include <tf/tf.h>
#include <tf/transform_listener.h>
#include <std_msgs/String.h>
#include <rosgraph_msgs/Log.h>
#include <ctime>
#include <cstdio>
#include <cstdlib>

void rosoutCB(const rosgraph_msgs::Log::ConstPtr& msg){
	if (msg->level == rosgraph_msgs::Log::ERROR){    
		printf("[test_rosout] FIND LOG ERR\n");
	}
}

int main(int argc, char** argv){
    ros::init(argc, argv, "test_rosout");

    ros::NodeHandle n;
    ros::Subscriber rosout_sb = n.subscribe("/rosout", 0, rosoutCB);


	ROS_INFO("this is a info");
	ros.spinOnce();
	ROS_ERROR("this is a err");
	ros.spinOnce();
	ROS_INFO("this is a info");
	ros.spinOnce();
	ROS_ERROR("this is a err");

	while (ros::ok()){
		ros::spin();
	}

    return 0;
}
