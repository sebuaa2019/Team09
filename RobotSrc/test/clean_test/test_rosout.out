[test_rosout] FIND LOG ERR
[test_rosout] FIND LOG ERR